// angular-bootstrap-md/tooltip/tooltip.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TooltipContainerComponent } from './tooltip.component';
import { TooltipDirective } from './tooltip.directive';
import { TooltipConfig } from './tooltip.service';
import { ComponentLoaderFactory } from '../utils/component-loader';
import { PositioningService } from '../utils/positioning';
var MDBTooltipModule = (function () {
    function MDBTooltipModule() {
    }
    MDBTooltipModule.forRoot = function () {
        return {
            ngModule: MDBTooltipModule,
            providers: [TooltipConfig, ComponentLoaderFactory, PositioningService]
        };
    };
    ;
    return MDBTooltipModule;
}());
export { MDBTooltipModule };
MDBTooltipModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                declarations: [TooltipDirective, TooltipContainerComponent],
                exports: [TooltipDirective],
                entryComponents: [TooltipContainerComponent]
            },] },
];
/** @nocollapse */
MDBTooltipModule.ctorParameters = function () { return []; };
