// angular-bootstrap-md/modals/modal.options.js
import { Injectable } from '@angular/core';
var ModalOptions = (function () {
    function ModalOptions() {
    }
    return ModalOptions;
}());
export { ModalOptions };
ModalOptions.decorators = [
    { type: Injectable },
];
/** @nocollapse */
ModalOptions.ctorParameters = function () { return []; };
var MDBModalRef = (function () {
    function MDBModalRef() {
    }
    /**
     * Hides the modal
     */
    MDBModalRef.prototype.hide = function () { };
    return MDBModalRef;
}());
export { MDBModalRef };
MDBModalRef.decorators = [
    { type: Injectable },
];
/** @nocollapse */
MDBModalRef.ctorParameters = function () { return []; };
export var modalConfigDefaults = {
    backdrop: true,
    keyboard: true,
    focus: true,
    show: false,
    ignoreBackdropClick: false,
    class: '',
    animated: true
};
export var ClassName = {
    SCROLLBAR_MEASURER: 'modal-scrollbar-measure',
    BACKDROP: 'modal-backdrop',
    OPEN: 'modal-open',
    FADE: 'fade',
    IN: 'in',
    SHOW: 'show' // bs4
};
export var Selector = {
    DIALOG: '.modal-dialog',
    DATA_TOGGLE: '[data-toggle="modal"]',
    DATA_DISMISS: '[data-dismiss="modal"]',
    FIXED_CONTENT: '.navbar-fixed-top, .navbar-fixed-bottom, .is-fixed'
};
export var TransitionDurations = {
    MODAL: 300,
    BACKDROP: 150
};
export var DISMISS_REASONS = {
    BACKRDOP: 'backdrop-click',
    ESC: 'esc'
};
